src/library/dlgtrackinfo.cpp

```diff
@@ -596,32 +596,23 @@ void DlgTrackInfo::slotKeyTextChanged() {
 
     // If the new key string is invalid and not empty them reject the new key.
     if (globalKey == mixxx::track::io::key::INVALID && !newKeyText.isEmpty()) {
-        txtKey->setText(KeyUtils::getGlobalKeyText(m_keysClone));
+        txtKey->setText(KeyUtils::getGlobalKeyText(m_keys));
         return;
     }
 
     // If the new key is the same as the old key, reject the change.
-    if (globalKey == m_keysClone.getGlobalKey()) {
+    if (globalKey == m_keys.getGlobalKey()) {
         return;
     }
 
     // Otherwise, accept.
-    m_keysClone = newKeys;
+    m_keys = newKeys;
 }
 
 void DlgTrackInfo::reloadTrackMetadata() {
     if (m_pLoadedTrack) {
-        // Allocate a temporary track object for reading the metadata.
-        // We cannot reuse m_pLoadedTrack, because it might already been
-        // modified and we want to read fresh metadata directly from the
-        // file. Otherwise the changes in m_pLoadedTrack would be lost.
-        TrackPointer pTrack(Track::newTemporary(
-                m_pLoadedTrack->getFileInfo(),
-                m_pLoadedTrack->getSecurityToken()));
```

Comment:
I think to fix the original bug, It is "just" required to copy the original bpm value over. 
or introduce a new populateFields() function which handles this. 
Like repopulateFieldsFromFile() or something.